parkbenchcommon/broadcastconsumer.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 Joel Allen Luellwitz and Emily Frost
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Provides the consuming component of a filesystem-based IPC mechanism."""
+
+import datetime
+import os
+import logging
+import time
+
+__all__ = ['BroadcastCheckError', 'BroadcastConsumer']
+
+SPOOL_PATH = '/var/spool'
+
+class BroadcastCheckError(Exception):
+    """This exception is raised when an error is encountered while attempting to check a
+    broadcast.
+    """
+
+class BroadcastConsumer(object):
+    """Provides the consuming component of a filesystem-based IPC mechanism.
+
+    program_name: The name of the program that issues this broadcast.
+    broadcast_name: The name of this broadcast.
+    minimum_delay: The minimum delay in seconds between broadcasts. If a second broadcast
+        occurs before minimum_delay has passed, the second broadcast is ignored.
+    """
+    def __init__(self, program_name, broadcast_name, minimum_delay):
+        self.logger = logging.getLogger(__name__)
+        self.logger.debug("Initializing consumer for broadcast %s from program %s.",
+                          broadcast_name, program_name)
+
+        self.broadcast_name = broadcast_name
+        self.program_name = program_name
+        self.broadcast_path = os.path.join(SPOOL_PATH, program_name, 'ramdisk', 'broadcast')
+        self.last_broadcast_time = datetime.datetime.now().isoformat()
+        self.next_check_time = time.time()
+        self.minimum_delay = minimum_delay
+
+
+        self.logger.info(
+            'The consumer for broadcast %s from program %s has been initialized.',
+            broadcast_name, program_name)
+
+    def check(self):
+        """Check if a new broadcast has been issued.
+
+        Returns True if a new broadcast has been issued. Returns False otherwise.
+        """
+        broadcast_updated = False
+
+        latest_broadcast_time = self._read_broadcast_time()
+
+        if latest_broadcast_time is not None:
+            if latest_broadcast_time > datetime.datetime.now().isoformat():
+                self.logger.warning('Read a broadcast from the future and ignoring it.')
+
+            else:
+                if self.next_check_time < time.time():
```

Comment:
I think the else for this condition should probably log a warning. Also, the condition should probably be reversed so the warning is first.

Reply:
We talked about this and this is a different feature. What you want is a warning if a broadcast is found (but not consumed) during the post-broadcast-consumption delay. It will require tracking two broadcast variables, `last_read_broadcast` and `last_consumed_broadcast`.